internal/bucket/lifecycle/lifecycle.go

```diff
@@ -311,6 +310,13 @@ func (lc Lifecycle) ComputeAction(obj ObjectOpts) Action {
 		}
 
 		if !rule.NoncurrentVersionExpiration.IsDaysNull() {
+			// Skip rules with newer noncurrent versions specified.
+			// These rules are not handled at an individual version
+			// level. ComputeAction applies only to a specific
+			// version.
+			if !obj.IsLatest && rule.NoncurrentVersionExpiration.NewerNoncurrentVersions > 0 {
```

Comment:
I doubt that we can set both NoncurrentDays and NewerNoncurrentVersions, can you double check ?

Reply:
I am positive. Find below an ILM policy snippet from https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketLifecycleConfiguration.html

>       <NoncurrentVersionExpiration>
>          <NewerNoncurrentVersions>integer</NewerNoncurrentVersions>
>          <NoncurrentDays>integer</NoncurrentDays>
>       </NoncurrentVersionExpiration>